### ISSUE TITLE
Undefined error when starting container when unable to downgrade

### ISSUE THREAD

**Issue by jason-nabooki** [tpetry/stackbricks-community#22]
When attempting to downgrade a version of a MySQL 8 database, an `undefined` error is thrown by Stackbricks. The container starts but promptly stops after it logs the error seen below.

**Steps to replicate**
1. Create MySQL 8.0.23 database
2. Start it up
3. Stop it once started
4. Change version to 8.0.29
5. Start it up
6. Stop it once started
7. Change version back to 8.0.23
8. Attempt to start

**Environment**
OSX 12.1 (21C52)
Docker App 4.9.0 (80466)

When Stackbricks eventually throws the `undefined` error. If you click `OK` and then try to start again, it then throws a `port already in use` error.

```
[Entrypoint] MySQL Docker Image 8.0.23-1.1.19
[Entrypoint] Starting MySQL 8.0.23-1.1.19
2022-06-12T23:01:29.320148Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.23) starting as process 24
2022-06-12T23:01:29.333899Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2022-06-12T23:01:29.822512Z 1 [ERROR] [MY-013171] [InnoDB] Cannot boot server version 80023 on data directory built by version 80029. Downgrade is not supported
mysqld: Can't open file: 'mysql.ibd' (errno: 0 - )
```

![image](https://user-images.githubusercontent.com/52981451/173465023-984e0ba6-0de8-4c49-805c-8ffe10e2fdcc.png)

![image](https://user-images.githubusercontent.com/52981451/173465061-e522de51-e5ad-4f39-9c99-2f27a945e7a9.png)

**Comment by tpetry** [comment]
MySQL changed the binary format of the data files in a patch version, as they don't follow semver and don't see why this is a problem. You can't effectively not downgrade anymore…

I have to look up the exact version they did the change (was it 8.0.25?) and hard-code the rule that downgrading is only possible withing the same binary format. I guess this will happen more often in the future…

**Comment by tpetry** [comment]
MySQL changed the on-disk file format with MySQL 8.0.29 (a patch release) making it impossible to downgrade from 8.0.29 fu any earlier release. The database version selection has been changed to reflect this and disallows downgrading in these cases. The bug is fixed with `0.5.1`.
